compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2514,6 +2514,17 @@ class ExtensionNullifiedByMember(method: Symbol, target: Symbol)(using Context)
        |
        |The extension may be invoked as though selected from an arbitrary type if conversions are in play."""
 
+class ExtensionHasDefault(method: Symbol)(using Context)
+  extends Message(ExtensionHasDefaultID):
+  def kind = MessageKind.PotentialIssue
+  def msg(using Context) =
+    i"""Extension method ${hl(method.name.toString)} should not have a default argument for its receiver."""
+  def explain(using Context) =
+    i"""Although extensions are ordinary methods, when they are invoked as a selection,
```

Comment:
nit: I think that the newline makes the message hard to parse now, since I first read only the first line, which is grammatically correct – `Although extensions are ordinary methods, when they are invoked as a selection,`

Reply:
It's not a complete sentence. However, I have rephrased to use no clauses.